rule/waitgroup-by-copy.go

```diff
@@ -0,0 +1,65 @@
+package rule
+
+import (
+	"github.com/mgechev/revive/lint"
+	"go/ast"
+)
+
+// WaitGroupByCopyRule lints sync.WaitGroup passed by copy in functions.
+type WaitGroupByCopyRule struct{}
+
+// Apply applies the rule to given file.
+func (r *WaitGroupByCopyRule) Apply(file *lint.File, _ lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+
+	onFailure := func(failure lint.Failure) {
```

Comment:
This is a very repetitive pattern. I'm wondering if it's possible to abstract this closure somehow.